agent/Retriever.py

```diff
@@ -0,0 +1,80 @@
+from llama_index import (
+    VectorStoreIndex,
+    SummaryIndex,
+    SimpleKeywordTableIndex,
+    SimpleDirectoryReader,
+    ServiceContext,
+)
+from llama_index.agent import OpenAIAgent
+from llama_index.schema import IndexNode
+from llama_index.tools import QueryEngineTool, ToolMetadata
+from llama_index.llms import OpenAI
+import os
+from llama_index.retrievers import RecursiveRetriever
+from llama_index.query_engine import RetrieverQueryEngine
+from llama_index.response_synthesizers import get_response_synthesizer
+
+class Retriever:
+    def __init__(self, index_path):
+        
+        self.llm = OpenAI(temperature=0, model="gpt-4-1106-preview")
+        self.service_context = ServiceContext.from_defaults(llm=llm)
+    
+    def load_documents(files=files):
+        self.docs = {}
+        for file in files.keys():
+            docs[file] = SimpleDirectoryReader(
+                input_files=[f"./{file}"],
+            ).load_data()
+
+
+
+    def create_agents(_files, _docs, _service_context):
+        self.agents = {}
+        for file in files.keys():
+            # build vector and summary index
+            vector_index = VectorStoreIndex.from_documents(docs[file], service_context=service_context)
```

Comment:
Parameters `_files`, `_docs`, and `_service_context` should be renamed without the preceding underscore if they are actually used, or removed if they are not.
    Additionally, instances of `files`, `docs`, and `service_context` should be prefixed with `self.` to indicate they are class attributes.
```suggestion
    def create_agents(self, files, docs, service_context):
```